erpnext/accounts/doctype/payment_entry/payment_entry.py

```diff
@@ -1889,20 +1931,26 @@ def set_paid_amount_and_received_amount(
 	return paid_amount, received_amount
 
 
-def apply_early_payment_discount(paid_amount, received_amount, doc):
+def apply_early_payment_discount(paid_amount, received_amount, doc, party_account_currency):
 	total_discount = 0
+	valid_discounts = []
 	eligible_for_payments = ["Sales Order", "Sales Invoice", "Purchase Order", "Purchase Invoice"]
 	has_payment_schedule = hasattr(doc, "payment_schedule") and doc.payment_schedule
+	is_multi_currency = party_account_currency != doc.company_currency
 
 	if doc.doctype in eligible_for_payments and has_payment_schedule:
 		for term in doc.payment_schedule:
 			if not term.discounted_amount and term.discount and getdate(nowdate()) <= term.discount_date:
```

Comment:
```suggestion
			if not term.discounted_amount and term.discount and doc.reference_date <= term.discount_date:
```

The relevant date for determining if the discount applies is the date the transaction was received. If this would make things easier, we could argue that the posting date would work as well – I guess they can be identical for most practical purposes.

Imagine the customer paid yesterday, but I'm recording the payment today. The discount should still apply, if yesterday was the last day of validity.

Reply:
I think reference date would be safest since it's purpose is to specify the actual cheque date or receipt date